### ISSUE TITLE
Relationship naming fallback in non-gender case

### ISSUE THREAD

**Issue by jkbecker** [monicahq/monica#3216]
**Is your feature request related to a problem? Please describe.**

I just came across the following case while importing my whole extended family and defining some relationships between them. If contacts don't have a gender defined, it seems like their relationship designation always defaults to the male version (brother, father, etc.). 

**Describe the solution you'd like**
It is super cumbersome for me to define genders of all my contacts just to not have my aunt show up as an uncle -- would it be possible for there to be a gender-neutral fallback for this case that is always correct? E.g., sibling instead of brother, parent instead of father, ..., etc.?

Not sure if this conflicts with the way it is currently implemented, but I feel like the current UI forces me to edit all my contacts one by one just to "correct" the gendered relationship for all female family members, instead of it being neutral until properly defined which is much less of a "pressure to clean up" because it's not "incorrect".

**Additional context**
References #287 but I felt this was slightly different in scope.

**Comment by tbirrell** [comment]
This is something that would require a bit of work behind the scenes because its using the translation files to display.

https://github.com/monicahq/monica/blob/24bd12b76d261604885ede8192c8719f03cc7441/resources/lang/en/app.php#L167-L170

This is the function that controls which version is returned:

https://github.com/monicahq/monica/blob/68ac2539711bcc1eff15e5b5c8eac11d25beee21/app/Models/Relationship/RelationshipType.php#L72-L116

So you could probably just add two lines to each gender-delineated translation group. Like:

     'relationship_type_parent' => 'parent',
     'relationship_type_parent_male' => 'father', 
     'relationship_type_parent_female' => 'mother', 
     'relationship_type_parent_with_name' => ':name’s parent', 
     'relationship_type_parent_male_with_name' => ':name’s father' 
     'relationship_type_parent_female_with_name' => ':name’s mother', 

But if you want to get neutral with the translation names, it will take a lot more work, and it will impact several database tables. Also for some term's you don't have common titles (in english at least), so it could be confusing.

    'relationship_type_nibling' => 'nibling',
    'relationship_type_male_nibling' => 'nephew',
    'relationship_type_nibling_female' => 'niece',
    'relationship_type_nibling_with_name' => ':name’s nibling',
    'relationship_type_nibling_male_with_name' => ':name’s nephew',
    'relationship_type_nibling_female_with_name' => ':name’s niece',


So in short, possible, but maybe not worth it? And will definitely need a fair bit of discussion depending how deep you wanna go.

**Comment by tbirrell** [comment]
Oh, and additionally, the vast majority of relationships don't have gender-specific terms, so it'll be a lot of redundancy, that may or may not be wanted. 

Though, this could be a good time to refactor and simplify this whole system. Which would definitely be a lot of work. 

https://github.com/monicahq/monica/blob/24bd12b76d261604885ede8192c8719f03cc7441/resources/lang/en/app.php#L119-L265

**Comment by jkbecker** [comment]
Oh, wow - I didn't expect it to have that much of an impact. :sweat_smile: 

I don't expect this to be integrated just based on my suggestion above, I assumed it was going to be a potential low hanging fruit. If a major refactoring of this sort is in the making, could we just add this to the backlog for things to consider?

**Comment by mnemonica** [comment]
@djaiss @asbiin: I'd love to work on this refactor: I use the hosted version of monicahq and it's frustrating to have my friend's nonbinary child be referred to as her "son".

Is having gender-neutral relationship labels (in addition to the usual gendered ones) something you'd be willing to have as a product feature? It'd be a big refactor (from a first-time committer, no less), so I don't want to spend a ton of time on technical details until I can be sure the maintainers would be open to such a change.

**Comment by asbiin** [comment]
@mnemonica Yes, we'd love to see a contribution like that, of course!
Don't hesitate if you have any trouble to do it.
Thank you!

**Comment by EasternPA** [comment]
I'm just trying this out and uncovered a whole bunch of things going on with registering my family and extended family. But this issue relates to the fact that gender is not maintained throughout the site -- or is maintained in an odd way. My aunt is not my sister's Uncle just like my sister is not my aunt's nephew. When adding family relationships, the relations that require gender for naming are just "off" a bit. spouse/wife, child/daughter, parent/mother, sibling/sister, grandparent/grandmother, the list goes on and on. Virtually every relation that is named based on the gender of the people involved is not straightforward throughout.

**Comment by virtualdxs** [comment]
Has there been any work on this? I'm tempted to as a workaround modify my translation files to remove gender entirely (e.g. changing "brother" and "sister" both to "sibling") but I would really appreciate proper gender handling.

**Comment by virtualdxs** [comment]
I suppose I probably should have checked the latest version before asking :smile: 

It seems like this is partially complete - neutral terms are the default, male terms have been added to at least the en lang, but AFAICT male terms aren't used right now, so should probably keep this open until that's handled

**Comment by djaiss** [comment]
Monica v2 will change the behaviour for this use case.
Maintaining relationship naming with all the gender use cases is extremely complex, especially from a localization (ie translation) point of view. 
In v2 (which isn't out yet), we let people define their own relationship types. So this problem becomes even harder if we want to support the gender dependencies of relationship names. We need to move away from this. So we've decided to simplify this entirely.

**Comment by Rustydw** [comment]
Hi, I don't seem to be able to link my friends who are in same-sex marriages unless I say Bob is Mark's spouse/wife. I've tried changing genders but still unable. Is this covered in Monica v2 as well? 
<img width="168" alt="image" src="https://user-images.githubusercontent.com/73714211/204061208-b2e13978-5b17-4748-b7a1-e1d9e3b0b568.png">
Or do you need someone to work thru the matrix of gender vs relationship name/type vs partner type possibilities?

**Comment by virtualdxs** [comment]
Spouse/wife is the correct option - essentially male terms seem to have been replaced with neutral ones, i.e. previously that would have said husband/wife. There seems to currently be no way to have it use the male terms.
